huaweicloud/services/secmaster/resource_huaweicloud_secmaster_indicator.go

```diff
@@ -0,0 +1,581 @@
+				ForceNew:    true,
+				Description: `Specifies the data source type.`,
+			},
+			"product_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the product name.`,
+			},
+			"product_feature": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the product feature.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceIndicatorCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createIndicator: Create a SecMaster indicator.
+	var (
+		createIndicatorHttpUrl = "v1/{project_id}/workspaces/{workspace_id}/soc/indicators"
+		createIndicatorProduct = "secmaster"
+	)
+	createIndicatorClient, err := cfg.NewServiceClient(createIndicatorProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating SecMaster client: %s", err)
+	}
+
+	createIndicatorPath := createIndicatorClient.Endpoint + createIndicatorHttpUrl
+	createIndicatorPath = strings.ReplaceAll(createIndicatorPath, "{project_id}", createIndicatorClient.ProjectID)
+	createIndicatorPath = strings.ReplaceAll(createIndicatorPath, "{workspace_id}", d.Get("workspace_id").(string))
+
+	createIndicatorOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json"},
+	}
+
+	createOpts, err := buildCreateIndicatorBodyParams(d, cfg)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	createIndicatorOpt.JSONBody = utils.RemoveNil(createOpts)
+	createIndicatorResp, err := createIndicatorClient.Request("POST", createIndicatorPath, &createIndicatorOpt)
+	if err != nil {
+		return diag.Errorf("error creating Indicator: %s", err)
+	}
+
+	createIndicatorRespBody, err := utils.FlattenResponse(createIndicatorResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("data.id", createIndicatorRespBody)
+	if err != nil {
+		return diag.Errorf("error creating Indicator: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+
+	return resourceIndicatorRead(ctx, d, meta)
+}
+
+func buildCreateIndicatorBodyParams(d *schema.ResourceData, cfg *config.Config) (map[string]interface{}, error) {
+	dataObject := map[string]interface{}{
+		"name":             d.Get("name"),
+		"indicator_type":   buildIndicatorTypeOpts(d.Get("type")),
+		"verdict":          d.Get("threat_degree"),
+		"data_source":      buildIndicatorDataSourceOpts(d.Get("data_source")),
+		"status":           d.Get("status"),
+		"confidence":       d.Get("confidence"),
+		"labels":           utils.ValueIngoreEmpty(d.Get("labels")),
+		"granular_marking": d.Get("granularity"),
+		"value":            d.Get("value"),
+		"environment":      buildEnvironmentOpts(d, cfg),
+		"defanged":         d.Get("invalid"),
+	}
+
+	if v, ok := d.GetOk("first_occurrence_time"); ok {
```

Comment:
I think d.GetOk is unnecessary because both `first_occurrence_time` and `last_occurrence_time` are required.

Reply:
sure